build.sbt

```
@@ -72,43 +67,53 @@ def freshProject(name: String, dir: File): Project = {
 // Fork each scala test for now, to work around persistent mutable state
 // in Rocket-Chip based generators
 def isolateAllTests(tests: Seq[TestDefinition]) = tests map { test =>
-      val options = ForkOptions()
-      new Group(test.name, Seq(test), SubProcess(options))
-  } toSeq
+  val options = ForkOptions()
+  new Group(test.name, Seq(test), SubProcess(options))
+} toSeq
 
 // Subproject definitions begin
-//
-// FIRRTL is handled as an unmanaged dependency. Make will build the firrtl jar
-// before launching sbt if any of the firrtl source files has been updated
-// The jar is dropped in chipyard's lib/ directory, which is used as the unmanagedBase
-// for all subprojects
-lazy val chisel  = (project in file("tools/chisel3"))
+
+val chiselVersion = "3.4.0"
+lazy val chisel = (project in file("tools/chisel3"))
+lazy val chiselLib = "edu.berkeley.cs" %% "chisel3" % chiselVersion
+// While not built from source, *must* be in sync with the chisel3 git submodule
+// Building from source requires extending sbt-sriracha or a similar plugin and
+//   keeping scalaVersion in sync with chisel3 to the minor version
+lazy val chiselPluginLib = "edu.berkeley.cs" % "chisel3-plugin" % chiselVersion cross CrossVersion.full
+
+lazy val firrtlRef = ProjectRef(file("tools/firrtl"), "firrtl")
+lazy val firrtlLib = "edu.berkeley.cs" %% "firrtl" % "1.4-SNAPSHOT"
```

```suggestion
lazy val firrtlLib = "edu.berkeley.cs" %% "firrtl" % "1.4.0"
```
I'd stick to stable versions when using published dependencies. Note that so long as you set `-Dsbt.sourcemode=true`, it's going to be using the source dependency so the version isn't used.

The `chiselVersion` matters a lot more because we can't build the plugin from source and use it here (limitation of SBT, not fundamental), so it's always going to be using the published plugin.

---

Sounds good. I've been changing alot of `X.X.SNAPSHOT` versions to `X.X.+` which should be the same thing (iirc)